### DISCUSSION TITLE
GitHub Code Search fails for MDX

### DISCUSSION THREAD

**Discussion by muendelezaji** [community/community#67012] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

It seems code search does not pick up MDX files in repos, despite being [defined](https://github.com/github-linguist/linguist/blob/7ca3799b8b5f1acde1dd7a8dfb7ae849d3dfb4cd/lib/linguist/languages.yml#L3936) in linguist.

For example, on [yewstack/yew](https://github.com/yewstack/yew), clicking MDX on the Languages section returns [no results](https://github.com/search?q=repo%3Ayewstack%2Fyew++language%3AMDX&type=code).

Likewise, a general [code search](https://github.com/search?q=language%3AMDX&type=code) through the whole of GitHub returns no results either.

Compare that with [search results](https://github.com/search?q=language%3Atsx&type=code) for a language like TSX.

Possibly similar to a past issue with V lang (https://github.com/github-linguist/linguist/issues/5320).

**Language bar**
<img width="318" alt="image" src="https://github.com/community/community/assets/1127221/b936a3d1-7214-4cba-9e5e-8f3dd7ea1877">

**Search results**
<img width="895" alt="image" src="https://github.com/community/community/assets/1127221/6950d0d9-4e1a-4897-b881-5d0b42ef91e0">

**Comment by spenserblack** [comment]
You might want to check [this troubleshooting section for linguist](https://github.com/github-linguist/linguist/blob/master/docs/troubleshooting.md#when-i-click-on-a-language-present-in-the-language-stats-bar-i-get-a-message-saying-your-search-did-not-match-any-code). This could be an issue with go-enry.

In fact, it looks like go-enry maps MDX to Markdown :thinking:
https://github.com/go-enry/go-enry/blob/731c88f59234d847ab86a8a8a2a1544485066d1e/internal/code-generator/generator/test_files/extension.gold#L650

**Reply by look** [reply]
BTW, thank you for doing the research on this. It was helpful in diagnosing the issue.

**Comment by look** [comment]
Thanks for the report. I'm tracking this bug in our backlog and have a plan to get it fixed. I'll report back when it's complete.

**Comment by look** [comment, marked as answer]
We have updated how we map language names and are now using the current version of Linguist's data for this. Searching for `language:MDX` is now fixed.

**Comment by muendelezaji** [comment]
I can confirm it works now. Thanks :)
